matrix-meetings-bot/src/service/GuestMemberService.ts

```diff
@@ -131,4 +133,89 @@ export class GuestMemberService {
       );
     }
   }
+
+  /**
+   * Changes power level of the guest user to default if guest is promoted by another user
+   *
+   * Change is active if the following is true:
+   *  - guest user power level change is enabled
+   *  - bot is allowed to change power levels
+   *
+   * @param roomId room id
+   * @param powerLevelEvent power level event
+   */
+  public async processPowerLevels(
+    roomId: string,
+    powerLevelEvent: IStateEvent<PowerLevelsEventContent>,
```

Comment:
@ahmadkadri We discussed this and you correctly have found that we should demote guests when the feature is enabled and default power level equal to users default. This change is applied. Please have a look.